SpatialGDK/Source/SpatialGDKTests/SpatialGDK/LoadBalancing/ReferenceCountedLockingPolicy/ReferenceCountedLockingPolicyTest.cpp

```diff
@@ -93,27 +126,164 @@ bool FWaitForActor::Update()
 	return (IsValid(Actor) && Actor->IsActorInitialized() && Actor->HasActorBegunPlay());
 }
 
-DEFINE_LATENT_AUTOMATION_COMMAND_FOUR_PARAMETER(FTestIsLocked, FAutomationTestBase*, Test, TSharedPtr<TestData>, Data, FName, Handle, bool, bExpected);
+DEFINE_LATENT_AUTOMATION_COMMAND_FOUR_PARAMETER(FAcquireLock, FAutomationTestBase*, Test, TSharedPtr<TestData>, Data, FName, ActorHandle, FString, DebugString);
+bool FAcquireLock::Update()
+{
+	AActor* Actor = Data->TestActors[ActorHandle];
+	const ActorLockToken Token = Data->LockingPolicy->AcquireLock(Actor, DebugString);
+
+	// If the token returned is valid, it MUST be unique
+	if (Token != SpatialConstants::INVALID_ACTOR_LOCK_TOKEN)
+	{
+		for (const TPair<AActor*, TArray<TPair<ActorLockToken, FString>>>& ActorLockingTokenAndDebugStrings : Data->TestActorToLockingTokenAndDebugStrings)
+		{
+			const TArray<TPair<ActorLockToken, FString>>& LockingTokensAndDebugStrings = ActorLockingTokenAndDebugStrings.Value;
+			bool TokenAlreadyExists = LockingTokensAndDebugStrings.ContainsByPredicate([Token](const TPair<ActorLockToken, FString>& Data)
+			{
+				return Token == Data.Key;
+			});
+			if (TokenAlreadyExists)
+			{
+				Test->AddError(FString::Printf(TEXT("AcquireLock returned a valid ActorLockToken that had already been assigned. Token: %d"), Token));
+			}
+		}
+	}
+
+	TArray<TPair<ActorLockToken, FString>>& ActorLockTokens = Data->TestActorToLockingTokenAndDebugStrings.FindOrAdd(Actor);
+	ActorLockTokens.Emplace(TPairInitializer<ActorLockToken, FString>(Token, DebugString));
+
+	return true;
+}
+
+DEFINE_LATENT_AUTOMATION_COMMAND_FOUR_PARAMETER(FReleaseLock, FAutomationTestBase*, Test, TSharedPtr<TestData>, Data, FName, ActorHandle, FString, LockDebugString);
+bool FReleaseLock::Update()
+{
+	const AActor* Actor = Data->TestActors[ActorHandle];
+
+	// Find lock token based on relevant lock debug string
+	FString& LockDebugStringRef = LockDebugString;
```

Comment:
This doesn't seem to be used.

Reply:
Yea my bad, this was me trying to do dumb things to get over the fact the capture for the lambda below was being weird, removed.